float32 centerX
float32 centerY
